src/main/java/frc/robot/subsystems/drivetrain/DrivetrainSubsystem.java

```diff
@@ -0,0 +1,138 @@
+package frc.robot.subsystems.drivetrain;
+
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.ctre.phoenix.motorcontrol.FollowerType;
+import com.ctre.phoenix.motorcontrol.InvertType;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.StatorCurrentLimitConfiguration;
+import com.ctre.phoenix.motorcontrol.SupplyCurrentLimitConfiguration;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import com.ctre.phoenix.motorcontrol.can.TalonFXConfiguration;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.ctre.phoenix.sensors.PigeonIMU;
+
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.geometry.Pose2d;
+import edu.wpi.first.wpilibj.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.kinematics.DifferentialDriveKinematics;
+import edu.wpi.first.wpilibj.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.wpilibj2.command.RamseteCommand;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.DrivetrainConstants;
+import frc.robot.Constants.RobotMap;
+  /**
+   * DrivetrainSubsystem handles all subsystem level logic for the drivetrain.
+   * Possibly also Ramsete idfk I haven't finished this class yet.
+   */
+public class DrivetrainSubsystem extends SubsystemBase {
+  private WPI_TalonFX m_leftMaster, m_rightMaster;
+  private WPI_TalonFX m_leftSlave, m_rightSlave;
+  private PigeonIMU m_pigeon;
+
+  private double m_yaw, m_pitch, m_roll;
+
+  private DifferentialDrive differentialdrive;
+
+  //Drivetrain kinematics, feed it width between wheels
+  private DifferentialDriveKinematics m_kinematics;
+
+  //Drivetrain odometry to keep track of our position on the field
+  private DifferentialDriveOdometry m_odometry;
+
+  private Pose2d m_pose;
+
+
+  public DrivetrainSubsystem() {
+    m_leftMaster = new WPI_TalonFX(RobotMap.kDrivetrainLeftBackTalonFX);
+    m_rightMaster = new WPI_TalonFX(RobotMap.kDrivetrainRightBackTalonFX);
+    m_leftSlave = new WPI_TalonFX(RobotMap.kDrivetrainLeftFrontTalonFX);
+    m_rightSlave = new WPI_TalonFX(RobotMap.kDrivetrainRightFrontTalonFX);
+
+    m_pigeon = new PigeonIMU(RobotMap.kPigeonIMU);
+
+    //Setting followers, followers don't automatically follow master's inverts so you must set the invert type to FollowMaster
+    m_leftSlave.follow(m_leftMaster, FollowerType.PercentOutput);
+    m_leftSlave.setInverted(InvertType.FollowMaster);
+    m_rightSlave.follow(m_rightMaster, FollowerType.PercentOutput);
+    m_rightSlave.setInverted(InvertType.FollowMaster);
+
```

Comment:
You will also need to invert the right side speed controllers. I don't see that yet.

Reply:
Fixed